### PR TITLE
Move program data files onto persistent storage

### LINKED ISSUE #4097: cache add: transferring cached image: Process exited with status 137 from signal KILL
**problem**
I am adding offline ~2G image to minikube with `cache add` command but it always failed.
When I ssh login VM I found a ~400M image file in `/tmp` folder. I try a lot of times it seems it is always failed at ~400M data transfered.
Small size image has no such problems.
The strange is when I put this image file in `files `folder it is always transferred to VM successfully. And also I can run `docker load -i xxx` successfully in VM.
**logs**
!   Failed to cache and load images: loading cached images: loading image {MINI_HOME}\cache\images\xxxx-web_2.89.27:
transferring cached image: Process exited with status 137 from signal KILL
**environment:**
windows 10 + virtualbox
minikube 1.0.0

### PR DESCRIPTION
**Pull request by tstromberg**
With this PR, `minikube start` is about 2 seconds slower on my MacBook Pro. There will be a follow-up PR that substantially increases performance by reusing these files rather than overwriting them each run, but I wanted to keep this PR on the smaller side.

## Notable Changes

- Binaries are stored in `/var/lib/minikube/binaries` (persistent)  rather than `/usr/bin`
- Images are stored in `/var/lib/minikube/images` (persistent)  rather than `/tmp`
- etcd data is stored in `/var/lib/minikube/etcd` instead of `/data/minikube` - leaving `/data` entirely for user data.
- apiserver timeout error now reveals if the process ever came online to assist with debugging.
- kubelet process is killed before upgrading the binary
- `sshrunner.Copy` no longer removes targets, leaving room for the future persistence PR. It's also been simplified using errgroup.
- Minor constants refactoring relating to affected paths. For example, constants referencing VM paths now have a "Guest" prefix:(`constants.AddonsPath -> constants.GuestAddonsDir`)

This PR closes #4097 and #3718
This PR helps with #3760 - but more work remains.

### CODE REVIEW

**Comment by medyagh** [comment]
this is amazing !

**Comment by medyagh** [comment]
does this PR work on windows as well ?

**Comment by medyagh** [comment]
one lint

```
pkg/minikube/bootstrapper/certs_test.go:66:2: undeclared name: `f` (typecheck)
	f.SetCommandToOutput(cmdMap)
	^
Makefile:308: recipe for target 'lint-ci' failed
make[1]: *** [lint-ci] Error 1
```

**Comment by medyagh** [comment]
the fail for none test seems to be related to this PR: as I haven't seen it else where:
```
* Aug 16 23:00:16 kvm-integration-slave systemd[1]: Stopped kubelet: The Kubernetes Node Agent.
* 
X Error getting machine logs: unable to fetch logs for: kube-scheduler
* 
* Sorry that minikube crashed. 
```

**Comment by tstromberg** [comment]
Current failures:

* **KVM_Linux**: `TestContainerd/GvisorRestart: Error configuring auth on host: OS type not recognized`
* **VirtualBox_Linux**: `TestVersionUpgrade: Wait failed: waiting for component=etcd: timed out waiting for the condition`
* **none_Linux**: `TestVersionUpgrade: Wait failed: waiting for k8s-app=kube-proxy: timed out waiting for the condition`

I suspect that these are flakes, but it worries me to see multiple TestVersionUpgrade failures, especially as the etcd folder has moved, so I will confirm locally.

**Comment by tstromberg** [comment]
The `none` issue appears to have persistence. VirtualBox passed, KVM failed with unrelated flake:
**TestDocker**: `Unable to start VM: detecting provisioner: OS type not recognized`
**TestContainerd/GvisorRestart**: `Temporary Error: Error configuring auth on host: OS type not recognized`
**TestStartStop/group/cni**: `Temporary Error: Error configuring auth on host: OS type not recognized`

File: pkg/minikube/constants/constants.go [hunk 1]
```diff
@@ -192,24 +192,42 @@ var ConfigFilePath = MakeMiniPath("config")
 // ConfigFile is the path of the config file
 var ConfigFile = MakeMiniPath("config", "config.json")
 
-// AddonsPath is the default path of the addons configuration
-const AddonsPath = "/etc/kubernetes/addons"
+// GetProfileFile returns the Minikube profile config file
+func GetProfileFile(profile string, miniHome ...string) string {
```
[the remark this answers is not in the document]
Comment (tstromberg) [reply pkg/minikube/constants/constants.go]: These unused functions must have been added by a bad merge. Removing.

**Comment by medyagh** [comment]
awww thanks @TravisBuddy

**Comment by tstromberg** [comment]
Unrelated flakes, merging.

**Pull request merged**